### DISCUSSION TITLE
A question on traps

### DISCUSSION THREAD

**Discussion by mmcheraghi** [stnolting/neorv32#1023] [Q&A] [answered]
Hi
Sorry I have a question about traps. when does a load access fault occur?

**Comment by NikLeberg** [comment]
Hi there @mahdi259 

The load access exception generally occures when either:
 - the neorv32 internal bus aborts an access with `err` (as opposed to `ack`): This can happen if for example the IMEM is implemented read-only but the CPU tried to write to it. But it can also be generated if an nonexistant address (in the _void_) is accessed. Or also it the memory alignment was not valid (reading a single byte from an address that only supports 32-bit/word accesses).
 - the external memory `xbus` aborts an access with `err`: IF you have attached something to the xbus, then it might generate an error if it is not accessed correctl.

For more information, see section [3.9. Traps, Exceptions and Interrupts](https://stnolting.github.io/neorv32/#_traps_exceptions_and_interrupts) in the datasheet.

If the problem persists, feel free to share your NEORV32 setup and we may point you to the some possible causes.

**Reply by mmcheraghi** [reply]
Sure, thanks a lot.

**Reply by mmcheraghi** [reply, marked as answer]
It seems that the problem was with memory sizes. It got `load access fault` trap when it returned from a function and while the `sp` of main function was reloaded. I increased each memory size to 1MB and it works without trap now!
Thanks a lot

**Reply by NikLeberg** [reply]
Glad to hear!
